petpals/messaging/routes.py

```diff
@@ -0,0 +1,193 @@
+            if message.conversation_id in unique_conversation_ids:
+                recent_messages.append(message)
+                unique_conversation_ids.remove(message.conversation_id)
+
+    return render_template(
+        'messages.html',
+        recent_messages=recent_messages,
+        current_user=current_user,
+        active_page=active_page,
+    )
+
+
+@router.get('/new')
+def new_message():
+    usernames = []
+    next_conversation_id = 0
+    conversation_id_query = db.session.query(func.max(Messages.conversation_id)).all()
+
+    if conversation_id_query[0][0] != None:
+        next_conversation_id = (
+            int(db.session.query(func.max(Messages.conversation_id)).all()[0][0]) + 1
+        )
+    else:
+        next_conversation_id = 1
+
+    users = User.query.all()
+    messages = Messages.query.all()
+
+    for user in users:
+        if current_user.username != user.username:
+            usernames.append(user.username)
+
+    for message in messages:
+        if current_user.username == message.sender_username:
+            if message.recipient_username in usernames:
+                usernames.remove(message.recipient_username)
+        elif current_user.username == message.recipient_username:
+            if message.sender_username in usernames:
+                usernames.remove(message.sender_username)
+
+    return render_template(
+        'new_message.html',
+        usernames=sorted(usernames),
+        sender=current_user.username,
+        active_page=active_page,
+        conversation_id=next_conversation_id,
+    )
+
+
+@router.post('/new/<recipient>')
+def send_new_message(recipient):
+    data = request.json
+
+    sender_checker = (
+        Messages.query.with_entities(
+            Messages.sender_username,
+            Messages.recipient_username,
+            Messages.conversation_id,
+        )
+        .filter_by(sender_username=data['sender'], recipient_username=data['recipient'])
+        .all()
+    )
+    recipient_checker = (
+        Messages.query.with_entities(
+            Messages.sender_username,
+            Messages.recipient_username,
+            Messages.conversation_id,
+        )
+        .filter_by(sender_username=data['recipient'], recipient_username=data['sender'])
+        .all()
+    )
+
+    new_message = Messages()
+
+    if db.session.query(func.max(Messages.conversation_id)).all()[0][0] == None:
+        new_message.conversation_id = 1
+    elif len(sender_checker) > 0:
+        new_message.conversation_id = sender_checker[0][2]
+    elif len(recipient_checker) > 0:
+        new_message.conversation_id = recipient_checker[0][2]
+    else:
+        new_message.conversation_id = (
+            int(db.session.query(func.max(Messages.conversation_id)).all()[0][0]) + 1
+        )
+
+    new_message.recipient_username = data['recipient']
+    new_message.sender_username = data['sender']
+    new_message.message = data['message']
+    new_message.time_sent = datetime.now()
```

Comment:
Should use the default time value.

Reply:
For some reason, the default value in the DB was not working, it would just be null every time.

Reply:
That's impossible because it should be not nullable. If you're talking about SQLAlchemy giving an error, it's because you need to set a default value in SQLAlchemy, you can look at the other models and their time variables as an example.